modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/ServerInfo.java

```diff
@@ -352,6 +355,8 @@ public static ServerInfo build(String name, String version) {
             }
 
             ComponentInfoImpl component = new ComponentInfoImpl(binfo, cname);
+            component.setRegistrationOrder(Long.valueOf(registrationOrder++));
```

Comment:
If you make `registrationOrder` a `long` then autoboxing will allow you to get rid of the `Long.valueOf` (another instance below for `bindex`/`setDeploymentOrder`)

Reply:
But what if i needed to be able to keep it null (unset), for legacy persisted distribs, typically?

Reply:
Well `Long.valueOf(registrationOrder++)` is never `null`, so you might as well take advantage of autoboxing here.

Reply:
This is on the live distrib (and the same interface is shared with persisted ones)

Reply:
(for now...)

Reply:
I don't get your point.
My point is that this:

    int foo = 1;
    component.setRegistrationOrder(Long.valueOf(foo++));

is identical too:

    long foo = 1;
    component.setRegistrationOrder(foo++);

Reply:
Ah! sorry got confused, will do